src/components/coin-icon/CoinIcon.js

```diff
@@ -1,62 +1,74 @@
-import React, { Fragment } from 'react';
+import { filter } from 'lodash';
+import React, { Fragment, useMemo } from 'react';
 import ReactCoinIcon from 'react-coin-icon';
-import { magicMemo } from '../../utils';
+import styled from 'styled-components/primitives';
 import CoinIconFallback from './CoinIconFallback';
 import CoinIconIndicator from './CoinIconIndicator';
-import { borders, colors } from '@rainbow-me/styles';
-import ShadowStack from 'react-native-shadow-stack';
+import { useColorForAsset, useTokenMetadata } from '@rainbow-me/hooks';
+import { RAINBOW_TOKEN_LIST } from '@rainbow-me/references/uniswap';
+import { isETH, magicMemo } from '@rainbow-me/utils';
 
 export const CoinIconSize = 40;
 
-const defaultShadow = [
-  [0, 4, 6, colors.dark, 0.04],
-  [0, 1, 3, colors.dark, 0.08],
-];
+const StyledCoinIcon = styled(ReactCoinIcon)`
+  background-color: ${({ color }) => {
+    // We have to give the outermost wrapper a background color
+    // or else React Native complains about shadow performance (and probably
+    // our perf is actually affected too).
+    // Read more about details of RN shadow / background requirements:
+    // 📎️ https://github.com/facebook/react-native/commit/e4c53c28aea7e067e48f5c8c0100c7cafc031b06
+    return color;
+  }};
+  opacity: ${({ isHidden }) => (isHidden ? 0.4 : 1)};
+`;
+
+// Create a flattened array containing all tokenAddresses and tokenSymbols for
+// the tokens with nice SVG icons from our `react-coin-icons` package
+const tokensWithIcons = filter(RAINBOW_TOKEN_LIST.tokens, 'extensions.color')
+  .map(({ address, symbol }) => [address, symbol])
+  .flat();
+
+const isIconAvailable = addressOrSymbol =>
+  tokensWithIcons.includes(addressOrSymbol);
 
 const CoinIcon = ({
   address,
-  bgColor,
   isHidden,
   isPinned,
-  showShadow = true,
   size = CoinIconSize,
-  shadow = defaultShadow,
-  symbol,
+  symbol = '',
   ...props
-}) =>
-  showShadow ? (
+}) => {
+  const tokenMetadata = useTokenMetadata(address);
+  const color = useColorForAsset({ address });
+
+  const forceFallback = useMemo(
```

Comment:
hey can you help me understand this logic, esp the part with the else of !isETH(address) -> !isETH(symbol)

Reply:
we deleted this logic so im resolvin this